src/EVM/Expr.hs

```diff
@@ -1219,8 +1219,11 @@ simplify e = if (mapExpr go e == e)
 simplifyProps :: [Prop] -> [Prop]
 simplifyProps ps = if cannotBeSat then [PBool False] else simplified
   where
-    simplified = remRedundantProps . map simplifyProp . flattenProps $ ps
-    cannotBeSat = isUnsat simplified
+    simplified = if (goOne ps == ps) then ps else simplifyProps (goOne ps)
+    cannotBeSat = PBool False `elem` simplified
+    goOne :: [Prop] -> [Prop]
+    goOne newPs = remRedundantProps . map simplifyProp . constPropagate . flattenProps $ newPs
```

Comment:
Would this work? Is this more or less readable?

```suggestion
    goOne = remRedundantProps . map simplifyProp . constPropagate . flattenProps
```

Reply:
Haha that's nice! I love this point-free version. I should do/learn point-free programming....